controllers/toolchainclusterhealth/toolchaincluster_healthcheck_controller_test.go

```diff
@@ -0,0 +1,295 @@
+		Get("healthz").
+		Persist().
+		Reply(200).
+		BodyString("unstable")
+	gock.New("http://not-found.com").
+		Get("healthz").
+		Persist().
+		Reply(404)
+
+	requeAfter := 10 * time.Second
+	withCA := false
+
+	t.Run("ToolchainCluster not found", func(t *testing.T) {
+		unstable, sec := newToolchainCluster("unstable", tcNs, "http://unstable.com", toolchainv1alpha1.ToolchainClusterStatus{})
+
+		cl := test.NewFakeClient(t, sec)
+		reset := setupCachedClusters(t, cl, unstable)
+		defer reset()
+		service := newToolchainClusterService(t, cl, withCA)
+		// given
+		controller, req := prepareReconcile(unstable, cl, service, requeAfter)
+
+		// when
+		_, err := controller.Reconcile(context.TODO(), req)
+
+		// then
+		require.NoError(t, err)
+
+	})
+
+	t.Run("Error while getting ToolchainCluster", func(t *testing.T) {
+		unstable, sec := newToolchainCluster("unstable", tcNs, "http://unstable.com", toolchainv1alpha1.ToolchainClusterStatus{})
+
+		cl := test.NewFakeClient(t, sec)
+
+		cl.MockGet = func(ctx context.Context, key runtimeclient.ObjectKey, obj runtimeclient.Object, opts ...runtimeclient.GetOption) error {
+			if _, ok := obj.(*toolchainv1alpha1.ToolchainCluster); ok {
+				return fmt.Errorf("mock error")
+			}
+			return cl.Client.Get(ctx, key, obj, opts...)
+		}
+
+		service := newToolchainClusterService(t, cl, withCA)
+		// given
+		controller, req := prepareReconcile(unstable, cl, service, requeAfter)
+
+		// when
+		_, err := controller.Reconcile(context.TODO(), req)
+
+		// then
+		require.EqualError(t, err, "mock error")
+
+	})
+	t.Run("if the connection cannot be established at beginning, then it should be offline", func(t *testing.T) {
+
+		stable, sec := newToolchainCluster("failing", tcNs, "http://failing.com", toolchainv1alpha1.ToolchainClusterStatus{})
+		cl := test.NewFakeClient(t, stable, sec)
+		service := newToolchainClusterService(t, cl, withCA)
+		// given
+		controller, req := prepareReconcile(stable, cl, service, requeAfter)
+
+		// when
+		_, err := controller.Reconcile(context.TODO(), req)
+
+		// then
+		require.NoError(t, err)
+		assertClusterStatus(t, cl, "failing", offline())
+	})
+	t.Run("if no zones nor region is retrieved, then keep the current", func(t *testing.T) {
+		stable, sec := newToolchainCluster("stable", tcNs, "http://cluster.com", withStatus(offline()))
+		cl := test.NewFakeClient(t, stable, sec)
+		resetCache := setupCachedClusters(t, cl, stable)
+		defer resetCache()
+		service := newToolchainClusterService(t, cl, withCA)
+		// given
+		controller, req := prepareReconcile(stable, cl, service, requeAfter)
+
+		// when
+		_, err := controller.Reconcile(context.TODO(), req)
+
+		// then
+		require.NoError(t, err)
+		assertClusterStatus(t, cl, "stable", healthy())
+	})
+
+	tests := map[string]struct {
+		tctype            string
+		apiendpoint       string
+		clustercondition1 toolchainv1alpha1.ToolchainClusterCondition
+		clustercondition2 toolchainv1alpha1.ToolchainClusterCondition
```

Comment:
what about using a list here?

```suggestion
		clusterConditions []toolchainv1alpha1.ToolchainClusterCondition
```

this would simplify a lot the `//then` checks

```diff
-			if k == "UnstableNoCondition" || k == "UnstableContainsCondition" {
-				assertClusterStatus(t, cl, tc.tctype, tc.clustercondition1, tc.clustercondition2)
-			} else {
-				assertClusterStatus(t, cl, tc.tctype, tc.clustercondition1)
-			}
+			assertClusterStatus(t, t, cl, tc.tctype, tc.clusterConditions...)
```

Reply:
Updated it